docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc

```diff
@@ -399,6 +401,53 @@ For example, if you work with Keycloak, you can use `keycloak.js` to authenticat
 </html>
 ----
 
+==== Use OIDC proxy
+
+Instead of tying your SPA to a specific OIDC provider by this OIDC provider's script, you may want to consider implementing an authorization code flow directly in the SPA by redirecting the users to a configurable OIDC authorization endpoint, accepting the OIDC redirect in the SPA and completing the authorization code flow by sending the code to a standard OIDC token endpoint and finaly sending the acquired access tokens to a Quarkus endpoint.
+
+Quarkus introduces an experimental `quarkus-oidc-proxy` OIDC proxy extension which provides an authorization code flow support for Quarkus OIDC `service` applications and can help to create SPAs by encapsulating all the OIDC provider configuration properties.
+
+All you need to do is to add the `quarkus-oidc-proxy` dependency to your Quarkus OIDC `service` application:
+
+[source,xml,role="primary asciidoc-tabs-target-sync-cli asciidoc-tabs-target-sync-maven"]
+.pom.xml
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-oidc-proxy</artifactId>
+</dependency>
+----
+
+[source,gradle,role="secondary asciidoc-tabs-target-sync-gradle"]
+.build.gradle
+----
+implementation("io.quarkus:quarkus-oidc-proxy")
+----
+
+Now, the SPA can use the OIDC proxy endpoints for authenticating the users using the authorization code flow and send the acquired access tokens to the Quarkus OIDC `service` application.
+
+For example, if the endpoint is listening at `http://localhost:8080`, then, after adding this dependency, the OIDC authorization endpoint is available at `http://localhost:8080/oidc/authorize` , the OIDC token endpoint is available at `http://localhost:8080/token`, the OIDC JWKS endpoint is available at `http://localhost:8080/oidc/jwks` and finally, the OIDC well known configuration endpoint which supports the discovery is available at `http://localhost:8080/.well-known/openid-configuration`. These endpoint paths can be customized.
+
+OIDC proxy endpoints must be public because they delegate to the real OIDC provider.
+
+For example, the following configuration is enabling an OIDC proxy over a Quarkus OIDC Auth0 `service` application, without having to configure your Auth0 application to allow redirects to the SPA page:
+
+[source,properties]
+----
+quarkus.oidc.auth-server-url=https://${auth0-dev}.us.auth0.com <1>
+quarkus.oidc.client-id=${auth0-client-id}
+quarkus.oidc.credentials.secret=${auth0-client-secret}
+
+quarkus.oidc.authentication.redirect-path=/oidc/callback <2>
+quarkus.oidc-proxy.external-redirect-uri=${external-spa-redirect-url}
+
+quarkus.http.auth.permission.oidcproxy.paths=/oidc/* <3>
```

Comment:
Well, if they must be public, why isn't this done by the extension?

Reply:
@FroMage Yes, good point, I've been contemplating about it but did not address, let me check

Reply:
@FroMage They are already public so that configuration was not needed, but added a note that if the service is already using a wildcard `/*` security policy then a permit exception for the proxy endpoints must be made